### DISCUSSION TITLE
add an ActionDispatch::IntegrationTest example to test helper examples

### DISCUSSION THREAD

**Discussion by mtoygar** [gonzalo-bulnes/simple_token_authentication#382] [Ideas]
test helper examples in the readme only uses ActionController::TestCase
https://github.com/gonzalo-bulnes/simple_token_authentication#testing

However currently Rails guides discourages the usage of `ActionController::TestCase` in favor of `ActionDispatch::IntegrationTest`. I believe adding an integration test example would be beneficial.

https://api.rubyonrails.org/classes/ActionController/TestCase.html

Below usage should be valid for tests that are inherited from ActionDispatch::IntegrationTest
```ruby
class SomeControllerTest < ActionDispatch::IntegrationTest
  test "index with token authentication via query params" do
    get some_route_url, params: { user_email: "alice@example.com", user_token: "1G8_s7P-V-4MGojaKD7a" }
    assert_response :success
  end

  test "index with token authentication via request headers" do
    get :some_route_url, headers: { 'X-User-Email' => 'alice@example.com', 'X-User-Token' => '1G8_s7P-V-4MGojaKD7a' }
    assert_response :success
  end
end
```

**Comment by gonzalo-bulnes** [comment]
Very good callout @mtoygar!

That README section seems obsolete now. (Thank you for the docs reference!)
Do you want to open a pull request to update it? I think your example can replace the existing one entirely, what do you think?
